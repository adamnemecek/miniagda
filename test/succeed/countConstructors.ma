-- 2010-01-13

data Nat : Set 
{ zero : Nat
; succ : Nat -> Nat
}

fun plus : Nat -> Nat -> Nat {}

mutual {
  fun f1 : Nat -> Nat
  { f1 zero = zero
  ; f1 (succ zero) = zero
  ; f1 (succ (succ n)) = g1 n
  }

  fun g1 : Nat -> Nat
  { g1 zero = zero
  ; g1 (succ n) = f1 (succ (succ n))
  }
}

mutual {
  fun f : Nat -> Nat
  { f zero = zero
  ; f (succ zero) = zero
  ; f (succ (succ n)) = g n
  }

  fun g : Nat -> Nat
  { g zero = zero
  ; g (succ n) = plus (f n) (plus (f (succ n)) (f (succ (succ n))))
  }
}

