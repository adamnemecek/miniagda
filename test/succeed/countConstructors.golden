MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "countConstructors.ma" ---
--- scope checking ---
--- type checking ---
type  Nat : Set
term  Nat.zero : < Nat.zero : Nat >
term  Nat.succ : ^(y0 : Nat) -> < Nat.succ y0 : Nat >
term  plus : Nat -> Nat -> Nat
{}
term  f1 : Nat -> Nat
term  g1 : Nat -> Nat
{ f1 Nat.zero = Nat.zero
; f1 (Nat.succ Nat.zero) = Nat.zero
; f1 (Nat.succ (Nat.succ n)) = g1 n
}
{ g1 Nat.zero = Nat.zero
; g1 (Nat.succ n) = f1 (Nat.succ (Nat.succ n))
}
term  f : Nat -> Nat
term  g : Nat -> Nat
{ f Nat.zero = Nat.zero
; f (Nat.succ Nat.zero) = Nat.zero
; f (Nat.succ (Nat.succ n)) = g n
}
{ g Nat.zero = Nat.zero
; g (Nat.succ n) = plus (f n) (plus (f (Nat.succ n)) (f (Nat.succ (Nat.succ n))))
}
--- evaluating ---
--- closing "countConstructors.ma" ---
